jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -28,24 +30,72 @@ class Catalogs(PluginTemplateMixin, ViewerSelectMixin):
     template_file = __file__, "catalogs.vue"
     catalog_items = List([]).tag(sync=True)
     catalog_selected = Unicode("").tag(sync=True)
+    from_file = Unicode().tag(sync=True)
+    from_file_message = Unicode().tag(sync=True)
     results_available = Bool(False).tag(sync=True)
     number_of_results = Int(0).tag(sync=True)
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        # TODO: Add more catalogs.
+
         self.catalog = SelectPluginComponent(self,
                                              items='catalog_items',
                                              selected='catalog_selected',
-                                             manual_options=['SDSS'])
+                                             manual_options=['SDSS', 'From File...'])
+
+        # file chooser for From File
+        start_path = os.environ.get('JDAVIZ_START_DIR', os.path.curdir)
```

Comment:
Maybe make a note that this is defined in `cli.py` in case we change that in the future?

Reply:
there already is a note in `cli.py` to that effect and this same logic is used in `data_tools.py`.  Do you think we need notes in the opposite direction (I just found it with a quick search-all)?

Reply:
Oh, if there is precedent, then nvm.